Add changelog for 4.4.0 (#4722)
thelounge/thelounge: CHANGELOG.md

<!-- New entries go after this line -->

## v4.4.0 - 2023-04-22

For more details, [see the full changelog](https://github.com/thelounge/thelounge/compare/v4.3.1...v4.4.0) and [milestone](https://github.com/thelounge/thelounge/milestone/42?closed=1).

> This is mostly a developer focused release. Max, Eric and others rewrote the whole thing in typescript / vue3,

Reviewer: ```suggestion
This is mostly a developer focused release. Max, Eric and others rewrote the whole thing in TypeScript / Vue 3,
```

which should make it much easier to add features and find bugs in the future. So huge kudos from the rest of the team!

Additionally, there's the obvious grab bag of fixes, dependency updates and improvements.  

> Settings are now grouped and hopefully easier to navigate for new users.

Reviewer: ```suggestion
Settings are now grouped and easier to navigate for new users.
```

> Following the nodejs maintenance schedule, The Lounge now needs at least node 14 to run.

Reviewer: ```suggestion
Following the Node.js maintenance schedule, The Lounge now needs at least Node.js 14 to run.
```

> A huge thanks to everyone who contributed in any way to this release, your help is much appreciated.

Reviewer: ```suggestion
A big thanks to everyone who contributed in any way to this release, your help is much appreciated.
```

you use `huge` earlier

Considering that a bunch of our dependencies had security issues assigned to them, all users are advised to update to the new version.

> Packagers: Considering the switch to Typescript, the server build now emits to the dist/ folder.  

Reviewer: ```suggestion
Packagers: Considering the switch to TypeScript, the server build now emits to the dist/ folder.  
```
